hyperspy/roi.py

```diff
@@ -374,6 +416,9 @@ def interactive(self, signal, navigation_signal="same", out=None,
                     'fft_shift', False):
                 raise NotImplementedError('ROIs are not supported when data '
                                           'are shifted during plotting.')
+        # Undefined if roi initialised without specifying parameters
+        if self._parameters_have_undefined():
```

Comment:
This can equivalent to `t.Undefined in list(self)`, which has the advantage:
- it is explicit and clear what it is.
- use standard python syntax
- don't add another method to these class

One issue with the current approach is that its behaviour is not consistent and misleading: `CircleROI` and `Line2D` don't check all values while other does.

Reply:
Agreed